src/main/resources/hudson/plugins/junitattachments/AttachmentTestAction/summary.jelly

```diff
@@ -13,7 +13,7 @@
 					<td class="pane">
 						<a class="${it.isImageFile(attachment) ? 'gallery' : ''}"
 						   title="${attachment}"
-						   href="attachments/${attachment}">${attachment}</a>
+						   href="${it.getUrl(attachment)}">${attachment}</a>
```

Comment:
Seems there is a manual test case (no JUnit test here in the plugin) confirming the fix at least for the case of `%`. It is not obvious if the fix will work for `<` or `&` or `"`, though I think it would.